src/libexpr/eval.cc

```diff
@@ -2296,6 +2317,57 @@ StorePath EvalState::coerceToStorePath(const PosIdx pos, Value & v, NixStringCon
     error("path '%1%' is not in the Nix store", path).withTrace(pos, errorCtx).debugThrow<EvalError>();
 }
 
+DerivedPath EvalState::coerceToDerivedPath(const PosIdx pos, Value & v, std::string_view errorCtx, bool validate)
```

Comment:
`validate` seems to be for unit testing only.

Is that really necessary? It's not good for coverage either.